enhancements/baremetal/upgrade-coreos-deploy-image.md

```diff
@@ -0,0 +1,275 @@
+---
+title: upgrade-coreos-deploy-image
+authors:
+  - "@zaneb"
+reviewers:
+  - "@hardys"
+  - "@dtantsur"
+  - "@elfosardo"
+  - "@sadasu"
+  - "@kirankt"
+  - "@asalkeld"
+  - "@cgwalters"
+  - "@cybertron"
+  - "@dhellmann"
+approvers:
+  - "@hardys"
+  - "@sadasu"
+creation-date: 2021-08-24
+last-updated: 2021-08-24
+status: implementable
+see-also:
+  - "/enhancements/coreos-bootimages.md"
+---
+
+# Upgrades of the CoreOS-based deploy image
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+To ensure that ironic-python-agent runs on top of an up-to-date OS, we will
+update the CoreOS image URLs in the baremetal Provisioning CR to the latest
+specified by the release metadata. For users running disconnected installs, we
+will require them to make the latest versions available and block further
```

Comment:
How often will the specified image change? We have goals of making platform upgrades low-risk and low-friction. If we regularly force disconnected users to update this RHCOS image just so they can apply a z-stream release, that extra step is significant enough to add noticeable friction. If we can ensure it's very easy to automate, that could help.

But do the goals of this proposal necessitate requiring the image to be kept exactly current? Or could you meet the goals while either:

- enabling the image to be updated but not requiring it
- enabling the image to the updated, but only requiring it to be within some version window. For example, you could insist that it have the same Y release.

Reply:
Good point, it's a shame we can't block only major upgrades. Although I don't think we tend to release more than 2-3 changes to the image per y stream.

On reflection, I think we should take advantage of the fact that we can set "Degraded=true" and "Upgradeable=False" independently:

* If the image URLs are missing altogether, we should set Upgradeable=False. This will prevent the user upgrading to to 4.11 (where there will be no ironic-ipa-downloader to fall back to) without setting them to _something_. We could potentially even remove this in 4.11.
* If the image URLs are out of date (or missing) we should set Degraded=true. This will make the user aware that they may be missing e.g. CVE fixes.

WDYT?

Reply:
What alerts will fire when an operator is degraded? Do we want them to fire for this case?